### ISSUE TITLE
nvim_buf_clear_highlight clears too many highlights from nvim_buf_set_virtual_text

### ISSUE THREAD

**Issue by blueyed** [neovim/neovim#9285]
`nvim_buf_clear_highlight` with id=2 will also clear the highlight for id=1.

```vim
let b = bufnr('%')

let id1 = nvim_buf_set_virtual_text(b, 0, 0, [['foo', 'Error']], {})
let id2 = nvim_buf_set_virtual_text(b, 0, 0, [['foo', 'Error']], {})

echo [id1, id2]

call nvim_buf_clear_highlight(b, id2, 0, -1)
```

This happens with both using `0` for an auto-generated src_id, or when using 1 and 2 directly.

Neovim v0.3.2-931-gdef608cd2.

**Comment by blueyed** [comment]
It appears that stacking is not supported for virtual text?!
https://github.com/neovim/neovim/blob/def608cd248218f9473caebb99e848e67a46a6f6/src/nvim/buffer.c#L5508-L5513

**Comment by bfredl** [comment]
Yes, it can be improved, but implementation will be more messy. If multiple ns has set the same line which one be first? The most recently added? Or by priority argument?

**Comment by blueyed** [comment]
`virt_text_src` could become a list maybe - or just like with normal highlights?
I think the priority should be the same as with highlights (likely the later one overrides the previous one).
